app/models/service/containers.rb

```diff
@@ -34,5 +33,21 @@ def self.destroy_wizard_state(wizard_state)
       wizard_state.destroy
       DockerContainerWizardState.destroy_all(["updated_at < ?", (Time.now - 24.hours)])
     end
+
+    def self.load_environment_variables(state, r)
```

Comment:
These methods are identical, could you refactor them so you can call something like ```build(:environment_variables, r)``` and ```build(:exposed_ports)```